Extend/Order_Controller.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend;
+
+use \OxidEsales\Eshop\Core\Registry;
+use \Wirecard\Oxid\Extend\Payment_Gateway;
+use \Wirecard\Oxid\Extend\Order;
+use \Wirecard\PaymentSdk\Response\FailureResponse;
+use \Wirecard\PaymentSdk\Response\InteractionResponse;
+use \Wirecard\PaymentSdk\Entity\Redirect;
+
+/**
+ * Class Order
+ *
+ * @package Wirecard\Extend
+ *
+ * @mixin \OxidEsales\Eshop\Application\Model\Order
+ */
+class Order_Controller extends Order_Controller_parent
+{
+    /**
+     * Extends the parent init function and finalizes the order in case it
+     * was a Wirecard payment method
+     */
+    public function init()
+    {
+        parent::init();
+        $oConfig = Registry::getConfig();
+        $sWdPaymentRedirect = $oConfig->getRequestParameter('wdpayment');
+        $oSession = Registry::getSession();
+        $sWdSessionToken = $oSession->getVariable('wdtoken');
+
+        if (!empty($sWdSessionToken) && $sWdSessionToken == $sWdPaymentRedirect) {
+            $sShopBaseUrl = $oConfig->getShopUrl();
```

Comment:
Maybe the inside of this IF could be some submethod?

Reply:
I don't think it is necessary to put it in a submethod